### ISSUE TITLE
How to deal with hover and selected component colors

### ISSUE THREAD

**Issue by lukasoppermann** [primer/primitives#363]
Hey @langermank and @josepmartins,

in the old setup we have used `color2k.darken` to create hover and selected colors.
We need to define how we want to deal with this now.

## Option 1: Use predefined color scales
I am assuming we want to avoid this as it creates unpredictable colors, correct? In this case one option would be to choose a color from the scale that is close to the darkened version.

## Option 2: Use overlay
Technically we could just add a second color on top, that is black with opacity. However this is also unpredictable and will probably create a lot of complications, so I would like to avoid this.

## Option 3: Continue with darken
If we are fine with have computer generated colors we could also continue with darken and add a darken transformer.
In this case however, we should define some requirements for a selected and hover color. Do we want to achieve a specific contrast to the existing color. Should it be darkened by a specific lum value? (We could try to use `HSLuv` here to get accurate decrease in lightness, as `hsl` is not very accurate).

**Comment by lukasoppermann** [comment]
> I'd say option 3 makes sense. On hover states the difference/jump in lightness is so subtle in some cases that using color scales would not be that precise.

Yes. One question is of course, if this minor change is still desired. We could also argue that a stronger change would make it easier to see when a hover or select happens.

> About the specific lum value, do you mean using the same value for all the hover variants in a Button component for instance?

For example. We could either specify that all hover colors should be darkened by a specific value, e.g. `.05` or that the difference between the default and hover color should be `1:1.5` (just making up random numbers here).
In general I think we need to define something, that would allow somebody who has not worked on primer before to define new colors.

**Comment by langermank** [comment]
I think this is something that will become more clear as we tackle functional color tokens. Right now we have custom state colors for component tokens with little consistency between each value. If we provided more generic `bg` functional tokens, we could assign stateful tokens to each of those values. As a very rough example:

```
--primer-color-bg-success
--primer-color-bg-success-hover
--primer-color-bg-success-active
```

Using `success` here instead of `primary` to denote that this isn't a `button` component token but moreso a functional token on top of the base scale that offers state. So basically.. I wonder if we can reach a place where its unlikely someone will need to add a component specific custom stateful token. Even the `action-list` tokens, which use alpha, could be generalized.

That said 😄 I think I would prefer these values be more tied to Figma in some way. If the JSON values sync up with Figma, couldn't we just use whatever hex value Figma spits out? This would require some trust in the designer (which is more often than not going to be a Primer designer) to use the right base scale value to start.

I think we should just establish these color manipulations as functional colors that have hex values from the start.

**Comment by lukasoppermann** [comment]
I like the idea of having tokens for all (most) the states and using a fixed hex value for that. However I don't get what you mean with this:

> That said 😄 I think I would prefer these values be more tied to Figma in some way. If the JSON values sync up with Figma, couldn't we just use whatever hex value Figma spits out?

Even with hex values we should have a defined way of creating a hover or active state to keep it consistent. This could be, reduce the color lightness by 10% or something else, but there should be a logic, so that we end up with similar colors and the knowledge (how to do it) is written down somewhere.

**Comment by langermank** [comment]
> Even with hex values we should have a defined way of creating a hover or active state to keep it consistent.

Totally agree with you! I like the idea of having written guidelines around that so it can be achieved in a color tool and not exclusively in code.

**Comment by lukasoppermann** [comment]
Closed by #492
